ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -12639,7 +12639,8 @@ static int zend_jit_fetch_obj(dasm_State          **Dst,
 		if (opline->opcode != ZEND_FETCH_OBJ_IS) {
 			|	SET_EX_OPLINE opline, REG0
 			if (opline->opcode != ZEND_FETCH_OBJ_W
-			 && (op1_info & MAY_BE_UNDEF)) {
+			 && (op1_info & MAY_BE_UNDEF)
+			 && opline->op1_type == IS_CV) {
```

Comment:
The same should be done for x86.
``ZEND_FETCH_OBJ_W`` may be followed at least by ``ASSIGN_DIM, ASSIGN_OBJ, FETCH_OBJ_W, FETCH_DIM_W``.
Did you check JIT for all these instructions?

After all, I'm not sure if this should be fixed at all. Targeting this to PHP-8.1 right before end of support cycle is too dangerous.

Reply:
@dstogov Sorry, I think I completely mixed this up in my head. Not `FETCH_OBJ_W`  needs to change but the opcodes that use it. I had no way to test this other than to see whether tests work...

> After all, I'm not sure if this should be fixed at all. Targeting this to PHP-8.1 right before end of support cycle is too dangerous.

I'll target `master` instead.